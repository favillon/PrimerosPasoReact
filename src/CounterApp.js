// Funtional components
import { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({value}) => {
    
    const [counter, setCounter] = useState(value);

    const handleAdd = () => {
        setCounter(counter + 1);
    }
    const handleReset = () => {
        setCounter(value);
    }
    const handleSubract = () => {
        setCounter(counter - 1);
    }
    
    return (
        <>
            <h1>CounterApp</h1>
            <p>{ counter }</p>
            <button onClick={ handleAdd }> +1 </button>
            <button onClick={ handleReset }> Reset </button>
            <button onClick={ handleSubract }> -1 </button>
        </>
    );
}

CounterApp.propType = {
    saludo : PropTypes.number
}
export default CounterApp;