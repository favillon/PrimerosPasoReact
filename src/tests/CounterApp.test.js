import React from "react";
//import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import CounterApp from "../CounterApp";


describe('Pruebas <CounterApp /> ', ()=>{

    test('Debe mostrar el counter', () =>{
        
        const wrapper = shallow(<CounterApp />);

        expect(wrapper).toMatchSnapshot();
    
    })
    test('Debe mostrar  el counter en 100', () =>{         
        const wrapper = shallow(<CounterApp value={100} />);

        const counterText = wrapper.find('p').text();
        
        expect(counterText).toBe('100');
    
    })
    test('Debe mostrar aumentar el counter en 101', () =>{         
        const wrapper = shallow(<CounterApp value={100} />);

        wrapper.find('button').at(0).simulate('click')
        
        const counterText = wrapper.find('p').text();
        expect(counterText).toBe('101');
    })

    test('Debe mostrar disminuir  el counter en 99', () =>{         
        const wrapper = shallow(<CounterApp value={100} />);

        wrapper.find('button').at(2).simulate('click')
        
        const counterText = wrapper.find('p').text();
        expect(counterText).toBe('99');
    })
    test('Debe mostrar resetear el counter', () =>{         
        const wrapper = shallow(<CounterApp value={50} />);

        wrapper.find('button').at(0).simulate('click')
        wrapper.find('button').at(0).simulate('click')
        
        const counterText1 = wrapper.find('p').text();
        //console.log(counterText1);
        expect(counterText1).toBe('52');

        wrapper.find('button').at(1).simulate('click')
        
        const counterText2 = wrapper.find('p').text();
        //console.log(counterText2);
        expect(counterText2).toBe('50');
    })
})
