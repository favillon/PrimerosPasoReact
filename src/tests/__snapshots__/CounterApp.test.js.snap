// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <CounterApp />  Debe mostrar el counter 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <p />
  <button
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    onClick={[Function]}
  >
     Reset 
  </button>
  <button
    onClick={[Function]}
  >
     -1 
  </button>
</Fragment>
`;
